query_base.go

```diff
@@ -146,6 +149,38 @@ func (q *baseQuery) GetTableName() string {
 	return ""
 }
 
+func (q *baseQuery) setComment(c string) {
+	q.comment = c
+}
+
+func (q *baseQuery) appendComment(ctx context.Context, b []byte) []byte {
+	// check if a comment was explicitly set on the query using the .Comment("...") api
+	if q.comment == "" {
+		// if not, check if one exists on the incoming context
+		if c, _ := ctx.Value(queryCommentCtxKey{}).(string); c == "" {
+			return b
+		} else {
+			q.comment = c
```

Comment:
Can you please change this so it does not modify the `baseQuery`? So the following is possible:

```
q := NewSelectQuery()
q.AppendQuery(ctxWithComment1) // one comment
q.AppendQuery(ctxWithComment2) // another comment
```

Also, one might think that both comments must be appended: one from the query and one from the context. I will leave it to you to decide what to do in such case.

Reply:
If multiple comments are allowed, then we can also consider supporting this: `NewSelect().Comment("comment1").Comment("comment2")`.

Reply:
> Can you please change this so it does not modify the `baseQuery`? So the following is possible:
> 
> ```
> q := NewSelectQuery()
> q.AppendQuery(ctxWithComment1) // one comment
> q.AppendQuery(ctxWithComment2) // another comment
> ```
> 
> Also, one might think that both comments must be appended: one from the query and one from the context. I will leave it to you to decide what to do in such case.

Ah, yes, good catch. Didn't think about the `AppendQuery` use case.

> If multiple comments are allowed, then we can also consider supporting this: NewSelect().Comment("comment1").Comment("comment2").

This seems reasonable, too. Let me experiment with the formatting a bit to see what looks best.